### PR TITLE
images/installer: Rewrite tectonic-installer for openshift-install

### PR DESCRIPTION
**Pull request by wking**
The golang-1.10 image has everything we need to build now.  I run the build in a golang-1.10 container using [`FROM ... AS ...`][1] like we used to.  But now I no longer install packages with `yum`.  And I use a recursive [`COPY`][2] to bring the built/fetched binaries over into the output container, which is based on `scratch`.

/assign @abhinavdahiya

[1]: https://docs.docker.com/engine/reference/builder/#from
[2]: https://docs.docker.com/engine/reference/builder/#copy

### CODE REVIEW

File: images/tectonic-installer/Dockerfile.ci [hunk 1]
```diff
@@ -1,25 +0,0 @@
-# This Dockerfile is a used as a CI job to validate Tectonic installer on CentOS
```
[the remark this answers is not in the document]
Comment (wking) [reply images/tectonic-installer/Dockerfile.ci]: grr.  I guess I'll leave that around then until we get openshift/release#1677 landed.

[the remark this answers is not in the document]
Comment (wking) [reply images/tectonic-installer/Dockerfile.ci]: > I guess I'll leave that around then...

Done with c7ea759 -> 7404847.

**Comment by wking** [comment]
Dropped the `docker.io` prefix I'd been using for testing with 7404847 -> 29e4d10.

**Comment by wking** [comment]
[The images failure][1] was:

```
Loading manifests from service-serving-cert-signer: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:e455472342e6554067046dbe6cd277f5fd0587cfaf56d21eaa54439b770439a5 ...
I0926 23:45:07.700595      22 logs.go:49] Unsolicited response received on idle HTTP channel starting with "HTTP/1.0 408 Request Time-out\r\nCache-Control: no-cache\r\nConnection: close\r\nContent-Type: text/html\r\n\r\n<html><body><h1>408 Request Time-out</h1>\nYour browser didn't send a complete request in time.\n</body></html>\n"; err=<nil>
error: unable to upload new layer (0): Patch https://registry.svc.ci.openshift.org/v2/ci-op-43zwvfh8/release/blobs/uploads/e26e5cfa-abc3-412a-8a3b-62d6484ce521?_state=EEFIl8MfVO7ZXUa97iSt1jwUd2cB7XZKYf9mJ3GASxx7Ik5hbWUiOiJjaS1vcC00M3p3dmZoOC9yZWxlYXNlIiwiVVVJRCI6ImUyNmU1Y2ZhLWFiYzMtNDEyYS04YTNiLTYyZDY0ODRjZTUyMSIsIk9mZnNldCI6MCwiU3RhcnRlZEF0IjoiMjAxOC0wOS0yNlQyMzo0NTowOS4yMDA3Njk4NjZaIn0%3D: operator "catalog" failed to map images: yaml: line 16: mapping values are not allowed in this context
Uploading ... 2018/09/26 23:45:10 Container release in pod release-latest failed, exit code 1, reason Error
```

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/343/pull-ci-openshift-installer-master-images/5/build-log.txt

**Comment by abhinavdahiya** [comment]
These seem like errors from `release-image` generation
```
2018/09/27 00:01:48 Executing release release-latest
2018/09/27 00:02:03 Copying artifacts from release-latest into /logs/artifacts/release-latest
info: Using registry public hostname registry.svc.ci.openshift.org
info: Found 73 images in image stream
info: Manifests will be extracted to /tmp/release-image-2018-09-27T000152Z087116184
warning: Could not load current user information: user: unknown userid 1137240000
Loading manifests from cluster-openshift-apiserver-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:770c03f88d9a947c9a1d3f29c5a0e313031dde86052fc5a151a99314e9d4fe6e ...
Loading manifests from cluster-kube-apiserver-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:12ba06fada065a12f8d2c3be27dcfa6577cb0a911886a5958a373f9e5d97b26f ...
Loading manifests from cluster-samples-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:bd3b11dacb5c4524dcd1e5c2814681174227d1b3e015aee99183e8fcd9d77fcc ...
Loading manifests from cluster-dns-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:4804588004bff91f5fa06bd515d72e746fc3bcd163eb0d2e0625e60c10514356 ...
Loading manifests from cluster-openshift-controller-manager-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:e34d27854162ac61e4a484c517651f5bd626c21d2dad126f91fe4ecd901f4529 ...
Loading manifests from catalog: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:bf9e1b9e0acbb056484a51fefe9dffb23bb7184e214cb48c344924fbe7f90091 ...
Loading manifests from cluster-image-registry-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:df10457dcee04dbea079253a67d97c233ef621ef0dd939c8a9bca03e849de846 ...
Loading manifests from cluster-version-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:6ed371948bfaf746dcb0263176687a625b1ef581447527cbb840b69c4aa7fa29 ...
Loading manifests from cluster-kube-controller-manager-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:0ce689191e2337e62f460a328f1aaf7d1ac12a80760793f73f96be7f7e5932b1 ...
Loading manifests from machine-config-operator: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:d2c602936193f81a808d477b065c01dab18d15914b2ec71c380f310d4a44b441 ...
Loading manifests from olm: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:43d39c9645cf54594f06fccd36a2ee294a9e6f7b4e09956f50fa1410a868e11e ...
Loading manifests from service-serving-cert-signer: registry.svc.ci.openshift.org/ci-op-43zwvfh8/stable@sha256:e455472342e6554067046dbe6cd277f5fd0587cfaf56d21eaa54439b770439a5 ...
error: unable to upload new layer (0): Patch https://registry.svc.ci.openshift.org/v2/ci-op-43zwvfh8/release/blobs/uploads/dcf4557c-9a87-4aa4-9d77-e3acdd113040?_state=gJbjHXXdyNlVUJgCRicnvCShiBRmwSEuX5WyHCwyJgt7Ik5hbWUiOiJjaS1vcC00M3p3dmZoOC9yZWxlYXNlIiwiVVVJRCI6ImRjZjQ1NTdjLTlhODctNGFhNC05ZDc3LWUzYWNkZDExMzA0MCIsIk9mZnNldCI6MCwiU3RhcnRlZEF0IjoiMjAxOC0wOS0yN1QwMDowMjowMS45ODg1MTQ2OTFaIn0%3D: operator "catalog" failed to map images: yaml: line 16: mapping values are not allowed in this context
Uploading ... 2018/09/27 00:02:03 Container release in pod release-latest failed, exit code 1, reason Error
2018/09/27 00:02:09 Ran for 23s
error: could not run steps: failed to wait for release pod to complete: could not wait for pod completion: the pod ci-op-43zwvfh8/release-latest failed after 14s (failed containers: release):  unknown
```
cc @smarterclayton

**Comment by wking** [comment]
For a similar error earlier, @smarterclayton pointed at [this indent issue][1].  Looks like that's currently still in master, but it could also be another busted JSON/YAML file causing the "operator "catalog" failed to map images: yaml: line 16: mapping values are not allowed in this context".

[1]: https://github.com/operator-framework/operator-lifecycle-manager/blob/c9e2936677618af6d054719655764a7b705ab136/manifests/image-references#L17

**Comment by wking** [comment]
More work on that indent issue ongoing in operator-framework/operator-lifecycle-manager#486.

**Comment by smarterclayton** [comment]
Goal is to have more gating in place in the next few days to mitigate the impact here - everything landing at once etc

**Comment by wking** [comment]
The images presublit it new today with openshift/release#1704.  Maybe we shoulfd roll it back temporarily?

File: images/installer/Dockerfile.ci [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# This Dockerfile is a used by CI to publish openshift/origin-v4.0:installer
+
+FROM openshift/origin-release:golang-1.10 AS build
+WORKDIR /go/src/github.com/openshift/installer
+COPY . .
+RUN hack/build.sh && hack/get-terraform.sh
```
Comment (wking) [inline images/installer/Dockerfile.ci]: We'll have to replace the `curl`-based `get-terraform.sh` here with a `yum` call or some such to comply with Red Hat's OCP Automated Release Policies.  But that's probably not too much of a rush.  I think `get-terraform.sh` should stay `curl`-based to be package-manager agnostic.

**Pull request merged**